src/map/utils/petutils.cpp

```diff
@@ -1266,12 +1266,11 @@ namespace petutils
             {
                 SetupPetWithMaster(PMaster, PPet);
             }
+
             // apply stats from previous zone if this pet is being transferred
             if (spawningFromZone)
             {
-                PPet->health.tp = static_cast<uint16>(static_cast<CCharEntity*>(PMaster)->petZoningInfo.petTP);
-                PPet->health.hp = static_cast<CCharEntity*>(PMaster)->petZoningInfo.petHP;
-                PPet->health.mp = static_cast<CCharEntity*>(PMaster)->petZoningInfo.petMP;
+                PPet->loadPetZoningInfo();
```

Comment:
@jmcmorris should pet level also be set here? I do set it in `loadPetZoningInfo` but was wondering why it wasn't set prior.

Reply:
No pet level shouldn't be set here as it is set when calculating the pet's stats. The only pets that uses it right now are jug pets.

Reply:
Removed pet level being set in loadPetZoningInfo